x/feeshare/ante/ante.go

```diff
@@ -62,29 +63,63 @@ type FeeSharePayoutEventOutput struct {
 	FeesPaid        sdk.Coins      `json:"fees_paid"`
 }
 
+func addNewFeeSharePayoutsForMsg(ctx sdk.Context, fsk FeeShareKeeper, toPay *[]sdk.AccAddress, m sdk.Msg) error {
+	if msg, ok := m.(*wasmtypes.MsgExecuteContract); ok {
```

Comment:
If the message contains another `authz.MsgExec` this logic doesn't work. You should add additional case for `authz.MsgExec`, and if it contains another `authz.MsgExec` message, call the function recursively.